File: 4. Week 4 - Pie Chart/solutions/r/visualization.R
# Visualize Time of Day Runs using ggplot2 (Pie Chart)

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

data_path <- "../../data/time_of_day_counts.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Calculate percentages for labels
  df$percent <- df$Count / sum(df$Count) * 100
  df$label <- paste0(df$Category, "\n", round(df$percent, 1), "%")
  
  # Plot
  # Pie charts in ggplot2 are Bar Charts + Polar Coordinates
  p <- ggplot(df, aes(x = "", y = Count, fill = Category)) +
    geom_bar(stat = "identity", width = 1) +
    
    # Convert to Pie
    coord_polar("y", start = 0) +
    
    # Custom Labels
    geom_text(aes(label = label), position = position_stack(vjust = 0.5)) +
    
    # Colors
    scale_fill_manual(values = c("AM" = "#9575CD", "PM" = "#673AB7", "Both" = "#D1C4E9")) +
    
    # Theme (Void removes axes/grid)
    theme_void() +
    labs(title = "Runs by Time of Day")
  
  ggsave("time_of_day_pie.png", plot = p, width = 8, height = 8)
  print("Chart saved to time_of_day_pie.png")

} else {
  print(paste("Error: File not found at", data_path))
}

File: 4. Week 4 - Pie Chart/solutions/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import os

def visualize():
    data_path = '../../data/time_of_day_counts.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    df = pd.read_csv(data_path)
    
    # Plot
    plt.figure(figsize=(8, 8))
    
    # Colors
    # Colors (Purple Theme)
    colors = ['#D1C4E9', '#9575CD', '#673AB7', '#311B92']
    
    plt.pie(df['Count'], labels=df['Category'], autopct='%1.1f%%', startangle=90, colors=colors)
    
    plt.title('Runs by Time of Day', fontsize=16)
    
    plt.tight_layout()
    plt.savefig('time_of_day_pie.png')
    print("Chart saved to time_of_day_pie.png")
    plt.show()

if __name__ == "__main__":
    visualize()
